cpp/src/parquet/statistics.cc

```diff
@@ -42,239 +45,332 @@ namespace parquet {
 
 template <typename DType, bool is_signed>
 struct CompareHelper {
-  typedef typename DType::c_type T;
+  using T = typename DType::c_type;
+
+  constexpr static T DefaultMin() { return std::numeric_limits<T>::max(); }
+  constexpr static T DefaultMax() { return std::numeric_limits<T>::lowest(); }
+  static T Coalesce(T val, T fallback) { return std::isnan(val) ? fallback : val; }
+
   static inline bool Compare(int type_length, const T& a, const T& b) { return a < b; }
+
+  static T Min(int type_length, T a, T b) { return a < b ? a : b; }
+  static T Max(int type_length, T a, T b) { return a < b ? b : a; }
 };
 
 template <>
 struct CompareHelper<Int96Type, true> {
-  static inline bool Compare(int type_length, const Int96& a, const Int96& b) {
+  using T = typename Int96Type::c_type;
+
+  constexpr static T DefaultMin() { return {}; }
+  constexpr static T DefaultMax() { return {}; }
```

Comment:
This doesn't look right.

Reply:
Good catch, forgot on my mental stack of things to fix...